Framework/DataHandling/src/SetSample.cpp

```diff
@@ -113,6 +115,22 @@ V3D cylBaseCentre(const std::vector<double> &cylCentre, double height,
   return V3D(cylCentre[0], cylCentre[1], cylCentre[2]) - halfHeight;
 }
 
+/**
+ * Return the centre coordinates of the base of a cylinder given the
+ * coordinates of the centre of the cylinder
+ * @param cylCentre Coordinates of centre of the cylinder (X,Y,Z) (in metres)
+ * @param height Height of the cylinder (in metres)
+ * @param axis The height-axis of the cylinder
+ */
+V3D cylBaseCentre(const std::vector<double> &cylCentre, double height,
+                  std::vector<double> axis) {
```

Comment:
Can we have `const std::vector<double>& axis` to avoid a copy? Same for the `axisXML` method below?